internal/shared/validations/validations.go

```diff
@@ -0,0 +1,71 @@
+package validations
+
+import (
+	"context"
+	"net/url"
+	"regexp"
+)
+
+// Valid git transports url
+
+var (
+	Transports = NewTransportSet(
+		"ssh",
+		"git",
+		"git+ssh",
+		"http",
+		"https",
+		"ftp",
+		"ftps",
+		"rsync",
+		"file",
+	)
+)
+
+// Checks for valid remote repository
+
+type TransportSet struct {
+	Transports map[string]struct{}
+}
+
+// NewTransportSet returns a TransportSet with the items keys mapped
+// to empty struct values.
+func NewTransportSet(items ...string) *TransportSet {
+	t := &TransportSet{
+		Transports: map[string]struct{}{},
+	}
+	for _, i := range items {
+		t.Transports[i] = struct{}{}
+	}
+	return t
+}
+
+// Valid returns true if transport is a known Git URL scheme and false
+// if not.
+func (t *TransportSet) Valid(transport string) bool {
+	_, ok := t.Transports[transport]
+	return ok
+}
+
+func CheckGitURL(gitURL string, ctx context.Context) bool {
+	u, err := url.Parse(gitURL)
+	if err == nil && !Transports.Valid(u.Scheme) {
+		return false
+	}
+	return true
+}
+
+func IsValidKubernetesNamespace(name string) bool {
```

Comment:
Please use the built in validation for k8s package. https://github.com/kubernetes/kubernetes/blob/2fb99eb9e81ec9fc92d8e694227772928f2eecc5/staging/src/k8s.io/pod-security-admission/admission/api/validation/validation.go#L73C20-L73C41

Reply:
The above method internally uses this method -https://github.com/kubernetes/kubernetes/blob/2fb99eb9e81ec9fc92d8e694227772928f2eecc5/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L188 

And as per the k8 docs.  **As a result, all namespace names must be valid [RFC 1123 DNS labels]**(https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names).   So I have modified it